src/did/src/gas.rs

```diff
@@ -27,7 +28,8 @@ pub struct EstimateGasRequest {
     pub max_priority_fee_per_gas: Option<U256>,
     pub gas: Option<U256>,
     pub value: Option<U256>,
-    pub data: Option<Bytes>,
+    #[serde(default, flatten)]
+    pub data: TransactionInput,
```

Comment:
serde attribute `flatten` is not supported by candid. I would expect test failures. Please add explicit candid/serde/storable tests to cover the possible encoding combinations